### ISSUE TITLE
Paket generate scripts - only generate "group" script

### ISSUE THREAD

**Issue by isaacabraham** [fsprojects/Paket#2189]
Given that since paket 4 the main group script doesn't rely on "child" scripts, it would be nice if you could just generate the top-level group script. Indeed, with such an approach you could completely do away with the folder structures then.

**Comment by forki** [comment]
// cc @smoothdeveloper

**Comment by smoothdeveloper** [comment]
@isaacabraham I need more info, I don't really understand without details, talk to me like I'm five :smile: 

AFAIK paket 3 also generated a standalone script per group, with same contents as we have in paket 4.

@isaacabraham are you looking for a `load.all.fsx` script which does same as `main.group.fsx` but also deals with other groups which exist in `paket.dependencies`?

**Comment by matthid** [comment]
@isaacabraham I think using only a single package out of an existing resolution is a valid use case for a script. On the other side I don't see a disadvantage in just using the complete group - even in those situation.

Maybe there are corner cases where this matters?

**Comment by isaacabraham** [comment]
@matthid absolutely, it's a valid use case.

I'm just thinking in terms of the simplest possible scenario (and probably the most common) - user opens a script and wants to start coding against their dependencies. Easiest thing is just (IMHO) ```#load ".paket\load\load.main.fsx"``` or whatever.

@smoothdeveloper All I'm saying is: have an option to *not* generate all individual package scripts and instead just the "all packages in the group" script. Almost like a "simple" mode which generates a single load file per group.

**Comment by smoothdeveloper** [comment]
@isaacabraham thanks I didn't understand that somehow, pardon my reading comprehension which isn't where it should at all times :)

In my use cases (repo with lots of dependencies) I generally use individual scripts on need basis.

What would you recommend to enable that option? It feels adding options to paket.dependencies and/or command line involve more complexity and futureproof constraints in the code base overall (those needs to pass through several layers AFAIU), the code is manageable but is not the most concise DSL to make adding options super easy / safe, and it implies more work for tooling integrating paket as well.

I'm just wondering what is the concern of having the indivudal scripts, we need to do the same resolution but it seems the implementation is not amortized.

I feel having the resolution amortized for both group and indivudal scripts is a better investment as I can see both use cases are probably spread among users of this feature.

Do you have an issue with the files or with the performance of generating the scripts?

**Comment by isaacabraham** [comment]
No, nothing like that. What is there works great now. It's just - again. IMHO - not the most intuitive for the first time user because you get a load of files - two per dependency (csx and fsx) - finding the "load all" script isn't that easy. I wonder whether promoting the "load alls" to the top folder would be an idea?

**Comment by smoothdeveloper** [comment]
We should enhance the documentation to promote better `#load @".paket/load/main.group.fsx"`

**Comment by isaacabraham** [comment]
Custom groups should also be placed at the top level though?

**Comment by smoothdeveloper** [comment]
@isaacabraham I'll look if bringing them in parent folder is trivial, although they will be mixed with a lot more files than and another breaking (minor) change :)

`#load ".paket/load/groupname/groupname.group.fsx"`

**Comment by isaacabraham** [comment]
no :-) sorry. what i meant was e.g.

```
.paket
   load
      net452
           main.group.fsx
           test.group.fsx
           release.group.fsx
           main
              |----all main deps
           test
              |----all test deps
           release
              |----all release deps
```

Does that make sense?
